pkg/apis/core/v1/validation/validation.go

```diff
@@ -18,19 +18,33 @@ package validation
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	"k8s.io/apimachinery/pkg/util/sets"
+	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/kubernetes/pkg/apis/core/helper"
 	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
 	apivalidation "k8s.io/kubernetes/pkg/apis/core/validation"
 )
 
-const isNegativeErrorMsg string = `must be greater than or equal to 0`
-const isNotIntegerErrorMsg string = `must be an integer`
+const (
+	isNegativeErrorMsg         string = `must be greater than or equal to 0`
+	isNotIntegerErrorMsg       string = `must be an integer`
+	maxParameterLength                = 100
```

Comment:
These should not be package-level consts - way too ambiguously named and not used.  I think you should just document this in your API spec and define a very local const in your validation